static/js/components/inputs/DateField.js

```diff
@@ -184,6 +186,7 @@ export default class DateField extends React.Component {
         fullWidth={true}
         value={edit.year !== undefined ? edit.year : ""}
         onChange={e => setNewDate(undefined, undefined, e.target.value)}
+        onBlur={onBlur}
```

Comment:
Should we also do this on the day and month fields? Though I'm not sure if it's possible to do it without causing weird behavior when the user moves from the day to month field for example